core/trino-main/src/main/java/io/trino/sql/gen/JoinCompiler.java

```diff
@@ -168,14 +179,17 @@ private LookupSourceSupplierFactory internalCompileLookupSourceFactory(List<Type
     {
         Class<? extends PagesHashStrategy> pagesHashStrategyClass = internalCompileHashStrategy(types, outputChannels, joinChannels, sortChannel);
 
```

Comment:
Maybe:
```
OptionalInt singleBigintJoinChannel = Optional.empty();
if (enableSingleChannelBigintLookupSource
    && joinChannels.size() == 1
    && types.get(getOnlyElement(joinChannels)) == BIGINT) {
  singleBigintJoinChannel = OptionalInt.of(getOnlyElement(joinChannels))
}
```
Then you don't need extra boolean where you need to figure out similar, but other variable name (`singleChannelBigint`)